### ISSUE TITLE
proposal: generalize toReadable

### ISSUE THREAD

**Issue by giacomociti** [zazuko/barnard59#199]
the base package has [`toReadable`](https://github.com/zazuko/barnard59/blob/master/packages/base/toReadable.js) operations (although they are not listed in `manifest.ttl`).

I propose a new implementation, simpler and more general:

```js
function toReadable() {
  return Readable.from(arguments)
}
```

with manifest entry:
```turtle
<toReadable> a p:Operation, p:ReadableObjectMode;
  rdfs:label "To Readable";
  rdfs:comment "Emits arguments as chunks.";
  code:implementedBy [ a code:EcmaScriptModule;
    code:link <node:barnard59-base/toReadable.js#default>
  ].
```

The proposed implementation is not limited to a single value:

```turtle
@prefix p: <https://pipeline.described.at/> .
@prefix base: <https://barnard59.zazuko.com/operations/base/> .
@prefix ntriples: <https://barnard59.zazuko.com/operations/formats/ntriples/> .
@prefix fs: <https://barnard59.zazuko.com/operations/rdf/fs.js#> .

<> a p:Pipeline ;
  p:steps [ 
    p:stepList
      (
        [ base:toReadable ("file1.ttl" "file2.ttl") ]
        [ fs:parse () ]
        [ ntriples:serialize () ]
        [ base:stdout () ]
      ) 
  ]
.
```

The same implementation works also with objects (even though I see less use cases):

```turtle
@prefix code: <https://code.described.at/> .
@prefix p: <https://pipeline.described.at/> .
@prefix base: <https://barnard59.zazuko.com/operations/base/> .

<> a p:Pipeline ;
  p:steps [ 
    p:stepList
      (       
        [ base:toReadable ("{ age: 34}"^^code:EcmaScript "{ age: 23}"^^code:EcmaScript) ]
        [ base:json\/stringify () ]
        [ base:stdout () ]
      ) 
  ]
.

```

**Comment by tpluscode** [comment]
That looks quite useful. Overall, a big 👍 

Two questions and two nits

1. Do you know if there is any different between `Readable.from` and the current implementation? Surely, it was created that way for a reason?
2. I would actually name the operation to match - `base:ReadableFrom ( "foo" "bar" )`
3. And maybe should suggest the use of `...args` over `arguments`
4. Does `"{ age: 34 }"^^code:EcmaScript` actually work to create an object from the JSON string? Neat

**Comment by tpluscode** [comment]
Ah, and one more thing. Would this still need two versions for `objectMode: true/false`?

**Comment by giacomociti** [comment]
`Readable.from` was added in node v12.3.0, v10.17.0 so I guess it was already available when the current implementation was made. I don't know why it is not taking advantage of it.

Renaming to `base:ReadableFrom` and using `...args` are good ideas, thanks.

Objects are parsed fine, and the readable stream is created in object mode. If I'm not wrong, the stream is in object mode also when using string arguments (this is also explicitly stated in the [documentation](https://nodejs.org/api/stream.html#streamreadablefromiterable-options)). Apparently it is working fine as a single function for both objects and strings (I tried to make separate functions but I got error when setting `objectMode` false for strings).

I can make a PR with the suggested changes so we can add some unit tests to ensure the behavior is correct.
